fulp_modules/LOOC_port/LOOC.dm

```diff
@@ -0,0 +1,90 @@
+GLOBAL_VAR_INIT(normal_looc_colour, "#ec0303")
+
+/client/verb/looc(msg as text)
+	set name = "LOOC"
+	set desc = "Local OOC, seen only by those in view."
+	set category = "OOC"
+
+	if(GLOB.say_disabled)	//This is here to try to identify lag problems
+		to_chat(usr, "<span class='danger'> Speech is currently admin-disabled.</span>")
+		return
+
+	if(!mob)
+		return
+
+	msg = copytext_char(sanitize(msg), 1, MAX_MESSAGE_LEN)
+	if(!msg)
+		return
+
+	if(!(prefs.chat_toggles & CHAT_OOC))
+		to_chat(src, "<span class='danger'> You have OOC muted.</span>")
+		return
+
+	if(!holder)
+		if(!GLOB.ooc_allowed)
+			to_chat(src, "<span class='danger'> OOC is globally muted</span>")
+			return
+		if(prefs.muted & MUTE_OOC)
+			to_chat(src, "<span class='danger'> You cannot use OOC (muted).</span>")
+			return
+		if(handle_spam_prevention(msg,MUTE_OOC))
+			return
+		if(findtext(msg, "byond://"))
+			to_chat(src, "<B>Advertising other servers is not allowed.</B>")
+			log_admin("[key_name(src)] has attempted to advertise in LOOC: [msg]")
+			return
+		if(mob.stat)
+			to_chat(src, "<span class='danger'>You cannot use LOOC while unconscious or dead.</span>")
+			return
+		if(isdead(mob))
+			to_chat(src, "<span class='danger'>You cannot use LOOC while ghosting.</span>")
+			return
+	if(is_banned_from(ckey, "OOC"))
+		to_chat(src, "<span class='danger'>You have been banned from OOC.</span>")
+		return
+	if(QDELETED(src))
+		return
+
+	msg = emoji_parse(msg)
+
+	if(SSticker.HasRoundStarted() && (msg[1] in list(".",";",":","#") || findtext_char(msg, "say", 1, 5)))
+		if(alert("Your message \"[msg]\" looks like it was meant for in game communication, say it in LOOC?", "Meant for LOOC?", "Yes", "No") != "Yes")
+			return
+
+	mob.log_talk(msg,LOG_OOC, tag="LOOC")
+
+	var/list/heard = get_hearers_in_view(7, src.mob.get_top_level_mob())
+	for(var/mob/M in heard)
+		if(!M.client)
+			continue
+		var/client/C = M.client
+		if (C in GLOB.admins)
+			continue //they are handled after that
+
+		if (isobserver(M))
+			continue //Also handled later.
+
+		if(C.prefs.chat_toggles & CHAT_OOC)
+			if(GLOB.LOOC_COLOR)
+				to_chat(C, "<font color='[GLOB.LOOC_COLOR]'><b><span class='prefix'>LOOC:</span> <EM>[src.mob.name]:</EM> <span class='message'>[msg]</span></b></font>")
+			else
+				to_chat(C, "<span class='looc'><span class='prefix'>LOOC:</span> <EM>[src.mob.name]:</EM> <span class='message'>[msg]</span></span>")
+
+	for(var/client/C in GLOB.admins)
+		if(C.prefs.chat_toggles & CHAT_OOC)
+			var/prefix = "(R)LOOC"
+			if (C.mob in heard)
+				prefix = "LOOC"
+			if(GLOB.LOOC_COLOR)
+				to_chat(C, "<font color='[GLOB.LOOC_COLOR]'><b>[ADMIN_FLW(usr)] <span class='prefix'>[prefix]:</span> <EM>[src.key]/[src.mob.name]:</EM> <span class='message'>[msg]</span></b></font>")
+			else
+				to_chat(C, "<span class='looc'>[ADMIN_FLW(usr)] <span class='prefix'>[prefix]:</span> <EM>[src.key]/[src.mob.name]:</EM> <span class='message'>[msg]</span></span>")
+
+
```

Comment:
This patch will enable ghosts to see LOOC in view range when OOC is enabled, and all LOOC messages when Ghost Whispers and OOC are enabled.

I actually tested this time.

```suggestion
	for(var/mob/M in GLOB.player_list)
		if(QDELETED(M))
			continue
		if(!isobserver(M))
			continue
		if(M.client in GLOB.admins)
			continue //Already handled
		if(!(M.client.prefs.chat_toggles & CHAT_OOC))
			continue
		if((!(M.client.prefs.chat_toggles & CHAT_GHOSTWHISPER)) && !(M in heard))
			continue //Piggyback on the ghost whispers preference to show/hide distant LOOC for ghosts
		
		var/link = FOLLOW_LINK(M, mob)
		if(GLOB.LOOC_COLOR)
			to_chat(M.client, "[link] <font color='[GLOB.LOOC_COLOR]'><b><span class='prefix'>LOOC:</span> <EM>[src.mob.name]:</EM> <span class='message'>[msg]</span></b></font>")
		else
			to_chat(M.client, "[link] <span class='looc'><span class='prefix'>LOOC:</span> <EM>[src.mob.name]:</EM> <span class='message'>[msg]</span></span>")

```

Reply:
cool im not 100% on ghosts being able to hear all looc though, it seems unnecessary

Reply:
You might want to get rid of the observer check on L66 then, so ghosts can hear LOOC in view.

Reply:
> cool im not 100% on ghosts being able to hear all looc though, it seems unnecessary

eh, sometimes there's some good banter or extra layers of story in there.

plus, the more people who can see and report LOOC abuse, the better, IMO.

Reply:
guess so il commit it and if it gets merge and becomes and issue i can always revert it